tailnet/test/integration/integration.go

```diff
@@ -111,13 +111,64 @@ type SimpleServerOptions struct {
 
 var _ ServerStarter = SimpleServerOptions{}
 
+type connManager struct {
+	mu    sync.Mutex
+	conns map[uuid.UUID]net.Conn
+}
+
+func (c *connManager) Add(id uuid.UUID, conn net.Conn) func() {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.conns[id] = conn
+	return func() {
+		c.mu.Lock()
+		defer c.mu.Unlock()
+		delete(c.conns, id)
+	}
+}
+
+func (c *connManager) CloseAll() {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	for _, conn := range c.conns {
+		_ = conn.Close()
+	}
+	c.conns = make(map[uuid.UUID]net.Conn)
+}
+
+type derpServer struct {
+	http.Handler
```

Comment:
Following on from your feedback - if we leave the `http.Handler` unnamed we only expose `Close` and `ServeHTTP`, which seems sensible.